Add documentation for nested selectors

silvenon opened the pull request:
The documentation for nested selectors was missing, so I read the relevant tests and comments to figure out how it works exactly and added the documentation. I'm differentiating between three types of nested selectors:

1. regular selectors
2. class selectors from `css`
3. component selectors from `styled`

The first two require a leading `&`, but component selectors don't work that way, the `&` must be omitted. This inconsistency should probably be fixed at some point.

This is a part of #219.

tkh44 commented on docs/nested.md:
```diff
@@ -1 +1,63 @@
+const paragraph = css`
+  color: gray;
+
+  & ${link} {
```
Does this really work? I think it would but you would need the `.` before the interpolation.

silvenon commented:
Ok, I added a sentence explicitly explaining why to add the dot `.` in case someone misses it when reading the code snippet. I also squashed everything into a single commit, I hope that's a desired practice.
